### PR TITLE
build: add automatic summary to PRs [skip release]

### PR DESCRIPTION
**Pull request by 3h4x**
# Summary of changes

# Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

## Dependencies

- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any teams which manage the dependencies have been notified of the breaking changes

## Bug (non-breaking change which fixes an issue):

- [ ] Have you **_written tests_** to protect against future occurrences of the bug?

## Feature (non-breaking change which adds functionality)

- [ ] Have you **_seen it working_** in a development environment?
- [ ] Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
- [ ] Have you implemented this feature as per the **_issue acceptance criteria_**?

### Substrate

If this change involves substrate, have you remembered:

- [ ] Storage Migration?
- [ ] RPC methods?
- [ ] Chainspec, both JSON specs & the module?
- [ ] Runtime versioning?
- [ ] Benchmarks?

## Breaking change (a feature that would cause existing functionality not to work as expected)

- [ ] Is the breaking change **_documented_**?
- [ ] Are your commits fully representative of the change?
- [ ] Has any previous functionality been **_deprecated_** and versioned?

## CI/CD

- [ ] If introducing new actions, have you **_looked at the action code_** for anything spurious?
- [ ] Have you written **_custom scripts_** for things that could be actions already on the marketplace?
- [ ] If introducing new actions, have you **_pegged a static version_**?

## Documentation Update

- [ ] Have you checked other documentation, such as the docs repository?
- [ ] If updating script documentation, have you **_tested it on both environments_**?

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

### Summary by OpenAI

Release Notes:

- Chore: Added a new GitHub Actions workflow.

> "A new workflow has arrived,
> No code changes, no need to strive.
> A chore it may be,
> But it brings us glee."

### CODE REVIEW

File: .github/workflows/auto-summary.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  cancel-in-progress: ${{ github.event_name != 'pull_request_review_comment' }}
+
+jobs:
+  summary:
+    name: PR Summary
+    runs-on: self-hosted
+    steps:
+      - uses: fluxninja/openai-pr-reviewer@0.54.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (3h4x) [inline .github/workflows/auto-summary.yml:26]: @openai what do you think about using secrets.GITHUB_TOKEN here?

File: .github/workflows/auto-summary.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  contents: read
+  pull-requests: write
+
+on:
+  pull_request:
+  pull_request_review_comment:
+    types: [created]
+
+concurrency:
+  group:
```
[the remark this answers is not in the document]
Comment (3h4x) [reply .github/workflows/auto-summary.yml:13]: @openai this concurrency group is 

```
    ${{ github.repository }}-${{ github.event.number || github.head_ref ||
    github.sha }}-${{ github.workflow }}-${{ github.event_name ==
    'pull_request_review_comment' && 'pr_comment' || 'pr' }}
```

explain like im 5

File: .github/workflows/auto-summary.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - uses: fluxninja/openai-pr-reviewer@0.54.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+        with:
+          debug: false
+          disable_review: false
+          review_simple_changes: false
+          review_comment_lgtm: true
```
[the remark this answers is not in the document]
Comment (3h4x) [reply .github/workflows/auto-summary.yml:32]: @openai its not needed here as gha is using gh api

**Comment by 3h4x** [comment]
@openai why didnt you comment lgtm on this pr?

File: .github/workflows/auto-summary.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+name: Automatic
+
+permissions:
+  contents: read
+  pull-requests: write
+
+on:
+  pull_request:
+  pull_request_review_comment:
+    types: [created]
```
[the remark this answers is not in the document]
Comment (3h4x) [reply .github/workflows/auto-summary.yml:10]: @openai what other events we could possibly need for action that does chatgpt summary and PR review?

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**